rust/agama-lib/src/base_http_client.rs

```diff
@@ -226,6 +226,16 @@ impl BaseHTTPClient {
         self.unit_or_error(response?).await
     }
 
+    /// Returns raw reqwest::Response. Use e.g. in case when response content is not
+    /// JSON body but e.g. binary data
+    pub async fn get_binary(&self, path: &str) -> Result<Response, ServiceError> {
```

Comment:
Does it only work with "binary" data? What if it is a text?

Reply:
binary here means that it doesn't try to "deserialize" the content into a particular format ... e.g. JSON like in case of the (most of) other calls.

Naming is not my strongest skill. May be it should be something like "get_raw_response" or so.

Reply:
Yes, *raw* sounds good to me.

Reply:
fine, so we have a winner